### PR TITLE
#166880067 Setup redux for web application

### PR DESCRIPTION
**Pull request by sebbss**
#### What does this PR do?

Sets up redux for the web application.

#### Description of Task to be completed?

- add a folder for actions
- add a folder for reducers
- add a store.js file

#### How should this be manually tested?
- clone this repo and run ``` npm install``` for dependencies.
- run ```npm start``` in your terminal 
- go to ```http://localhost:8080``` in your browser and inspect using react devtools 
- you will be able to see the store of the application

### Screenshots.
image

#### What are the relevant pivotal tracker stories?

[#166880067](https://www.pivotaltracker.com/story/show/166880067)

### CODE REVIEW

**Review by NicholusMuwonge — CHANGES_REQUESTED** [review]
Hey Joseph, I appreciate the work put out, but I feel like the demonstration of the task is still lacking. I would recommend adding something that changes state to fully show the functionality of the redux you just used is valid otherwise the current testing process only shows react implementation not redux.

**Comment by sebbss** [comment]
> Hey Joseph, I appreciate the work put out, but I feel like the demonstration of the task is still lacking. I would recommend adding something that changes state to fully show the functionality of the redux you just used is valid otherwise the current testing process only shows react implementation not redux.

@NicholusMuwonge I appreciate your concern but the fact that you see the store means that I have setup redux for application. things like changing state will happen when we are using the redux itself.

**Comment by NicholusMuwonge** [comment]
> > Hey Joseph, I appreciate the work put out, but I feel like the demonstration of the task is still lacking. I would recommend adding something that changes state to fully show the functionality of the redux you just used is valid otherwise the current testing process only shows react implementation not redux.
> 
> @NicholusMuwonge I appreciate your concern but the fact that you see the store means that I have setup redux for application. things like changing state will happen when we are using the redux itself.

@sebbss, The only reason that I am requesting for the addition is to ascertain if the store will work perfectly in storing the states even with changes. In this scenario, we eliminate the possibility of any feature bugs arising from this particular implementation. so kindly consider a more detailed demonstration. Kindly, please.

**Comment by sebbss** [comment]
> > > Hey Joseph, I appreciate the work put out, but I feel like the demonstration of the task is still lacking. I would recommend adding something that changes state to fully show the functionality of the redux you just used is valid otherwise the current testing process only shows react implementation not redux.
> > 
> > 
> > @NicholusMuwonge I appreciate your concern but the fact that you see the store means that I have setup redux for application. things like changing state will happen when we are using the redux itself.
> 
> @sebbss, The only reason that I am requesting for the addition is to ascertain if the store will work perfectly in storing the states even with changes. In this scenario, we eliminate the possibility of any feature bugs arising from this particular implementation. so kindly consider a more detailed demonstration. Kindly, please.

image
this now shows change in state @NicholusMuwonge

**Review by frankopkusianwar — APPROVED** [review]
LGTM

**Comment by NicholusMuwonge** [comment]
> > > > Hey Joseph, I appreciate the work put out, but I feel like the demonstration of the task is still lacking. I would recommend adding something that changes state to fully show the functionality of the redux you just used is valid otherwise the current testing process only shows react implementation not redux.
> > > 
> > > 
> > > @NicholusMuwonge I appreciate your concern but the fact that you see the store means that I have setup redux for application. things like changing state will happen when we are using the redux itself.
> > 
> > 
> > @sebbss, The only reason that I am requesting for the addition is to ascertain if the store will work perfectly in storing the states even with changes. In this scenario, we eliminate the possibility of any feature bugs arising from this particular implementation. so kindly consider a more detailed demonstration. Kindly, please.
> 
> image
> this now shows change in state @NicholusMuwonge

Thanks for replying to that feedback @sebbss I think this demonstrates the functionality alright.

**Review by NicholusMuwonge — APPROVED** [review]

**Review by richien — APPROVED** [review]

File: src/__test__/Home.test.js [hunk 1]
```diff
@@ -1,11 +1,16 @@
...
 
 
 describe('App', () => {
   it('should match snapshot', () => {
     const wrapper = shallow(<Home />);
     expect(wrapper.find('h1').text()).toBe('Welcome to Authors Haven');
   });
+  it('should match snapshot', () => {
+    const wrapper = shallow(<Login />);
+    expect(wrapper.find('h1').text()).toBe('login here');
```
Comment (MuhanguziDavid) [inline src/__test__/Home.test.js]: Use this for snapshot tests `expect(wrapper).toMatchSnapshot();`

File: src/containers/Hello.js [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+  }
+}
+
+export const mapStateToProps = state => ({
+  index: state.index,
+});
+export default connect(
+  mapStateToProps,
+  { test: FirstAction },
+)(Hello);
```
Comment (MuhanguziDavid) [inline src/containers/Hello.js:34]: Ensure to have `propTypes` for your components/containers e.g. have a `test` in your `propTypes`

**Review by MuhanguziDavid — CHANGES_REQUESTED** [review]
Nice work with setting up redux, please test your actions and reducers and enusre to have the rest of the components tested appropriately.

File: src/containers/Hello.js (hunk 2 above)
Comment (sebbss) [reply src/containers/Hello.js:34]: i have implemented this feedback

File: src/__test__/Home.test.js (hunk 1 above)
Comment (sebbss) [reply src/__test__/Home.test.js]: I have implemented this feedback

**Pull request merged**